### PR TITLE
VK/ORLP-165 checking of file extension

### PR DESCRIPTION
**Pull request by KuryloVolodymyr**
Functionality to check the type of file was added. Now user can upload only images

### CODE REVIEW

File: src/main/java/com/softserve/academy/spaced/repetition/service/ImageService.java [hunk 1]
```diff
@@ -51,10 +52,16 @@ public Image addImageToDB(MultipartFile file) throws ImageRepositorySizeQuotaExc
             throw new MultipartException("File upload error: file is too large.");
         } else {
             String base64 = encodeToBase64(file);
-            String imageType = file.getContentType();
-            image = new Image(base64, imageType, user, fileSize);
-            imageRepository.save(image);
-            image = imageRepository.getImageWithoutContent(image.getId());
+            if (!file.getContentType().split("/")[0].equals("image"))
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/service/ImageService.java]: According to JavaDoc https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/multipart/MultipartFile.html#getContentType--
... null if not defined...
It means that a NullPointerException is possible here.

File: src/main/java/com/softserve/academy/spaced/repetition/service/ImageService.java [hunk 2]
```diff
@@ -51,10 +52,16 @@ public Image addImageToDB(MultipartFile file) throws ImageRepositorySizeQuotaExc
...
+            {
+                throw new FileIsNotAnImageException();
+            }
+            else {
+                String imageType = file.getContentType();
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/service/ImageService.java]: Reuse this variable "imageType" in your check.

File: src/main/java/com/softserve/academy/spaced/repetition/logger/logs/log.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+2017-09-28 22:17:28,661 [WARN ][    logger.Log4jImpl] - Exception in service = com.softserve.academy.spaced.repetition.service.ImageService@76837082, method = Image com.softserve.academy.spaced.repetition.service.ImageService.addImageToDB(MultipartFile)
+2017-09-28 22:17:28,663 [WARN ][    logger.Log4jImpl] - Exception in controller = com.softserve.academy.spaced.repetition.controller.ImageController@22d9f079, method = ResponseEntity com.softserve.academy.spaced.repetition.controller.ImageController.addImageToDB(MultipartFile)
+2017-09-30 21:42:24,990 [WARN ][    logger.Log4jImpl] - Exception in service = com.softserve.academy.spaced.repetition.service.ImageService@1f0a62b3, method = Image com.softserve.academy.spaced.repetition.service.ImageService.addImageToDB(MultipartFile)
+2017-09-30 21:42:24,991 [WARN ][    logger.Log4jImpl] - Exception in controller = com.softserve.academy.spaced.repetition.controller.ImageController@4284caa4, method = ResponseEntity com.softserve.academy.spaced.repetition.controller.ImageController.addImageToDB(MultipartFile)
```
Comment (b0hdan) [inline src/main/java/com/softserve/academy/spaced/repetition/logger/logs/log.txt]: I suppose as soon as you delete this file it'll be okay. Try just reverting changes.

File: src/main/java/com/softserve/academy/spaced/repetition/service/ImageService.java [hunk 4]
```diff
@@ -52,9 +54,18 @@ public Image addImageToDB(MultipartFile file) throws ImageRepositorySizeQuotaExc
...
             String imageType = file.getContentType();
-            image = new Image(base64, imageType, user, fileSize);
-            imageRepository.save(image);
-            image = imageRepository.getImageWithoutContent(image.getId());
+            if (imageType != null && !imageType.split("/")[0].equals("image"))
+            {
+                throw new FileIsNotAnImageException();
+            }
+            else if(imageType == null){
+                throw new NullPointerException();
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/service/ImageService.java]: Who will handle this exception?

File: src/main/java/com/softserve/academy/spaced/repetition/service/ImageService.java [hunk 5]
```diff
@@ -52,12 +54,18 @@ public Image addImageToDB(MultipartFile file) throws ImageRepositorySizeQuotaExc
         } else {
             String base64 = encodeToBase64(file);
             String imageType = file.getContentType();
-            image = new Image(base64, imageType, user, fileSize);
-            imageRepository.save(image);
-            image = imageRepository.getImageWithoutContent(image.getId());
-        }
-        return image;
+            if (imageType == null || !imageType.split("/")[0].equals("image"))
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/service/ImageService.java]: Seems like formatting if off here...

**Pull request merged**